mso/resource_mso_fabric_policies_synce_interface_policy.go

```diff
@@ -0,0 +1,315 @@
+package mso
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/ciscoecosystem/mso-go-client/client"
+	"github.com/ciscoecosystem/mso-go-client/container"
+	"github.com/ciscoecosystem/mso-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceMSOSyncEInterfacePolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMSOSyncEInterfacePolicyCreate,
+		Read:   resourceMSOSyncEInterfacePolicyRead,
+		Update: resourceMSOSyncEInterfacePolicyUpdate,
+		Delete: resourceMSOSyncEInterfacePolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceMSOSyncEInterfacePolicyImport,
+		},
+
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"template_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 64),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"uuid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"admin_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"enabled", "disabled",
+				}, false),
+			},
+			"sync_state_msg": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"enabled", "disabled",
+				}, false),
+			},
+			"selection_input": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"enabled", "disabled",
+				}, false),
+			},
+			"src_priority": {
+				Type:         schema.TypeInt,
+				Computed:     true,
+				Optional:     true,
+				ValidateFunc: validation.IntBetween(1, 254),
+			},
+			"wait_to_restore": {
+				Type:         schema.TypeInt,
+				Computed:     true,
+				Optional:     true,
+				ValidateFunc: validation.IntBetween(0, 12),
+			},
+		},
+	}
+}
+
+func setSyncEInterfacePolicyData(d *schema.ResourceData, msoClient *client.Client, templateId, policyName string) error {
+
+	response, err := msoClient.GetViaURL(fmt.Sprintf("api/v1/templates/%s", templateId))
+	if err != nil {
+		return err
+	}
+
+	policy, err := GetPolicyByName(response, policyName, "fabricPolicyTemplate", "template", "syncEthIntfPolicies")
+	if err != nil {
+		return err
+	}
+
+	d.SetId(fmt.Sprintf("templateId/%s/SyncEInterfacePolicy/%s", templateId, models.StripQuotes(policy.S("name").String())))
+	d.Set("template_id", templateId)
+	d.Set("name", models.StripQuotes(policy.S("name").String()))
```

Comment:
Small suggestion so StripQuotes is only called once for name.
```suggestion
    name := models.StripQuotes(policy.S("name").String())
	d.SetId(fmt.Sprintf("templateId/%s/SyncEInterfacePolicy/%s", templateId, name))
	d.Set("template_id", templateId)
	d.Set("name", name)
```